Funcky.Analyzers/Funcky.Analyzers/SyntaxNodeExtensions.cs

```diff
@@ -0,0 +1,78 @@
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Funcky.Analyzers;
+
+internal static class SyntaxNodeExtensions
+{
+    // Copied from Roslyn's source code as this API is not public:
+    // https://github.com/dotnet/roslyn/blob/232f7afa4966411958759c880de3a1765bdb28a0/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs#L925
+    public static bool IsInExpressionTree(
+        [NotNullWhen(returnValue: true)] this SyntaxNode? node,
+        SemanticModel semanticModel,
+        [NotNullWhen(returnValue: true)] INamedTypeSymbol? expressionTypeOpt,
+        CancellationToken cancellationToken)
+    {
+        if (expressionTypeOpt != null)
+        {
+            for (var current = node; current != null; current = current.Parent)
+            {
+                if (current.IsAnyLambda())
```

Comment:
you avoid the switch-expression here because of the fall through, but you could extract it into a function.